Test suite fails for scitokens-1.6.2 with pyjwt-1.6.4: str vs decode mismatch

The test suite for scitokens-1.6.2 is failing when run against EPEL packages, namely pyjwt-1.6.4 because `SciToken.serialize()` is recording `_serialized_token` as a `str`, while `SciToken.discover(...)` stores `_serialized_token` as a `bytes`.

This should have been caught by the minimal dependencies test in the GHA workflow, but the `requirements.txt` file was not updated as part of #123.

```
$ PYTHONWARNINGS="" python3 -m pytest tests/ -vvvv -x --lf
========================================= test session starts =========================================
platform linux -- Python 3.6.8, pytest-2.9.2, py-1.4.32, pluggy-0.3.1 -- /usr/bin/python3
cachedir: .cache
run-last-failure: rerun last 10 failures
rootdir: /home/duncan.macleod/git/scitokens, inifile:
plugins: xdist-0.0.0, mock-1.10.4, cov-2.5.1
collected 45 items

tests/test_create_scitoken.py::TestCreation::test_discover FAILED

============================================== FAILURES ===============================================
_____________________________________ TestCreation.test_discover ______________________________________

self = <test_create_scitoken.TestCreation testMethod=test_discover>

    def test_discover(self):
        """
            Test wlcg bearer token discovery
            """
        # unset any wlcg discovery environment variables
        try:
           del os.environ['BEARER_TOKEN']
        except KeyError:
           pass
        try:
           del os.environ['BEARER_TOKEN_FILE']
        except KeyError:
           pass
        try:
           del os.environ['XDG_RUNTIME_DIR']
        except KeyError:
           pass

        # move any /tmp/bt_u$ID file out of the way
        try:
            bt_file = 'bt_u{}'.format(os.geteuid())
        except AttributeError as exc:  # windows doesn't have geteuid
            self.skipTest(str(exc))
        bt_path = os.path.join('/tmp', bt_file)
        (bt_fd, bt_tmp) = tempfile.mkstemp()
        os.close(bt_fd)
        if os.path.isfile(bt_path):
            os.rename(bt_path, bt_tmp)

        # check that the function fails properly
        with self.assertRaises(IOError):
            print(self._token.discover())

        # generate a token and save it as /tmp/bt_u$ID
        tmp_file_token = scitokens.SciToken(key = self._private_key, key_id="tmp_file")
        tmp_file_token['scope'] = 'tmp_file'
        tmp_file_token_s = tmp_file_token.serialize(issuer="local")
        with open(bt_path, 'w') as f:
            f.write(tmp_file_token_s.decode('utf-8'))

        # discover a token and check we found /tmp/bt_u$ID
        token = self._token.discover(public_key = self._public_pem)
>       self.assertEqual(token._serialized_token, tmp_file_token._serialized_token)
E       AssertionError: 'eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtp[533 chars]fXSw' != b'eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImt[534 chars]fXSw'

tests/test_create_scitoken.py:342: AssertionError
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

--- a/src/scitokens/scitokens.py
+++ b/src/scitokens/scitokens.py
@@ -123,7 +123,7 @@ class SciToken(object):
         :param str issuer: A string indicating the issuer for the token.  It should be an HTTPS address,
                            as specified in https://tools.ietf.org/html/draft-ietf-oauth-discovery-07
         :param int lifetime: Number of seconds that the token should be valid
-        :return str: base64 encoded token
+        :return bytes: base64 encoded token
         """
 
         if include_key is not False:
@@ -164,6 +164,8 @@ class SciToken(object):
             encoded = jwt.encode(payload, self._key, algorithm = self._key_alg, headers={'kid': self._key_id})
         else:
             encoded = jwt.encode(payload, self._key, algorithm = self._key_alg)
+        if isinstance(encoded, bytes):  # pyjwt < 2 returns bytes
+            encoded = encoded.decode("utf-8")
         self._serialized_token = encoded
 
         # Move claims over to verified claims
@@ -175,9 +177,7 @@ class SciToken(object):
 
         # Encode the returned string for backwards compatibility.
         # Previous versions of PyJWT returned bytes
-        if not isinstance(encoded, bytes):
-            encoded = str.encode(encoded)
-        return encoded
+        return str.encode(encoded, encoding="utf-8")
 
     def update_claims(self, claims):
         """

